File: 217_contains_duplicate/containsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> hashSet = new HashSet<Integer>();

        for (int num : nums) {
            if (hashSet.contains(num)) {
                return true;
            }

            hashSet.add(num);
        }
        return false;
    }
}

File: 217_contains_duplicate/containsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, bool> counter;

        for (int num : nums) {
            if (counter.find(num) != counter.end()) return true;
            counter[num] = true;
        }
        
        return false;
    }
};